core/contracts/financial-templates/expiring-multiparty/PricelessPositionManager.sol

```diff
@@ -406,9 +406,17 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
         fees()
         nonReentrant()
     {
-        require(_checkCollateralization(collateralAmount, numTokens), "CR below GCR");
-
         PositionData storage positionData = positions[msg.sender];
+
+        // Either the new create ratio or the resultant position CR must be above the current GCR.
+        require(
+            (_checkCollateralization(
+                _getFeeAdjustedCollateral(positionData.rawCollateral).add(collateralAmount),
+                positionData.tokensOutstanding.add(numTokens)
+            ) || _checkCollateralization(collateralAmount, numTokens)),
```

Comment:
nit: if it doesn't increase the bytecode size (that's a big if :)), do you think this would be a little more readable if we separated this into intermediate variables?

```
bool newTokensAboveGCR = ...;
bool resultingPoistionAboveGCR = ...;
require(newTokensAboveGCR || resultingPoistionAboveGCR, "Must meet GCR")
```

Reply:
this will definitely increase the bytecode but would make it easier to read. Curious how much it increase it by.

Reply:
I can experiment

Reply:
This increased bytecode by 30

Reply:
```
 // Either the new create ratio or the resultant position CR must be above the current GCR.
        bool resultantPositionCRGreaterThanGCR = _checkCollateralization(
            _getFeeAdjustedCollateral(positionData.rawCollateral).add(collateralAmount),
            positionData.tokensOutstanding.add(numTokens)
        );
        bool creationCRGreaterThanGCR = _checkCollateralization(collateralAmount, numTokens);
        require(
            (resultantPositionCRGreaterThanGCR || creationCRGreaterThanGCR),
            "New CR below GCR"
        );
```

Reply:
Not super opinionated here, but I'd probably err on the side of saving bytecode and making it a little less readable, just for the sake of making things easier down the road.

Reply:
agreed